src/vs/workbench/contrib/chat/browser/chatDragAndDrop.ts

```diff
@@ -122,12 +125,19 @@ export class ChatDragAndDrop extends Themable {
 		}
 	}
 
-	private getAttachContext(e: DragEvent): IChatRequestVariableEntry[] {
+	private async getAttachContext(e: DragEvent): Promise<IChatRequestVariableEntry[]> {
 		switch (this.getActiveDropType(e)) {
 
 			case ChatDragAndDropType.FILE: {
 				const data = extractEditorsDropData(e);
-				const contexts = data.map(editorInput => getEditorAttachContext(editorInput));
+				const contexts = await Promise.all(data.map(async editorInput => {
+					if (editorInput.resource && /\.(png|jpg|jpeg)$/i.test(editorInput.resource.path)) {
+						const fileBuffer = await this.fileService.readFile(editorInput.resource);
```

Comment:
A bit suprised to see a `readFile` call in here. Would it not be possible to pass the resource as attachement like we do with files and have it be read later on?

Reply:
https://github.com/microsoft/vscode/pull/228706#discussion_r1762260699

i guess kind of similar to this question here. since we are creating the attachment, would we want to create the attachment with the correct type and id and such (for image, it will be the buffer/uint8array) or have the extension convert it? i'm not sure what is best

Reply:
I'm not sure to be honest. I think ideally it would be the same as text files. We don't read those and save the contents in `value` so it should also not be done for images in theory. However, I assume this would not work for dnd from the web? If that is the case then supporting two different ways of image contexts would also not be great...